utils/filesystem/filesystem.go

```diff
@@ -42,12 +52,12 @@ func NewFs(fsType FilesystemType) FS {
 	return NewStandardFileSystem()
 }
 
-// Converts file system error into common errors
-func ConvertFileSytemError(err error) error {
+// ConvertFileSystemError converts file system error into common errors
```

Comment:
I know this isn't necessarily from this change, but I think for discovery reasons this should be in commonerrors. That is where I assume all the convert to common errors stuff it

Reply:
each module has its own convertError function because the errors are specific to their domain. Here to do with files systems, other places to do with hashing, etc.. If we put them in the commonerrors, it means everytime the commonerror module is loader (almost everywhere) all the errors and modules imported in that module would also be loaded. meaning every user of common loaded would load os, hash, ... even if not used. Having them in the module they are related to means there is a separation of concern. The fact that the context error are converted in the commonerrors is only because context is pretty much used everywhere too but we could have created a context module